modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -40,45 +40,187 @@ sealed trait ElasticRequest[A]
+   *
+   * @param index
+   *   the name of the index to create the document in
+   * @param doc
+   *   the document to be created, represented by an instance of type `A`
+   * @tparam A
+   *   the type of the document to be created. An implicit `Schema` instance must be in scope for this type
+   * @return
+   *   an instance of [[CreateRequest]] that represents the create operation to be performed.
+   */
   final def create[A: Schema](index: IndexName, doc: A): CreateRequest =
     Create(index = index, document = Document.from(doc), refresh = None, routing = None)
 
+  /**
+   * Constructs an instance of [[CreateWithIdRequest]] used for creating a document with specified ID in the specified
+   * index.
+   *
+   * @param index
+   *   the name of the index to create the document in
+   * @param id
+   *   the ID of the new document
+   * @param doc
+   *   the document to be created, represented by an instance of type `A`
+   * @tparam A
+   *   the type of the document to be created. An implicit `Schema` instance must be in scope for this type
+   * @return
+   *   an instance of [[CreateRequest]] that represents the create with id operation to be performed.
+   */
   final def create[A: Schema](index: IndexName, id: DocumentId, doc: A): CreateWithIdRequest =
     CreateWithId(index = index, id = id, document = Document.from(doc), refresh = None, routing = None)
 
+  /**
+   * Constructs an instance of [[CreateIndexRequest]] used for creating an empty index.
+   *
+   * @param name
+   *   the name of the index to be created
+   * @return
+   *   an instance of [[CreateIndexRequest]] that represents the create index operation to be performed.
+   */
   final def createIndex(name: IndexName): CreateIndexRequest =
     CreateIndex(name = name, definition = None)
 
+  /**
+   * Constructs an instance of [[CreateIndexRequest]] used for creating an index with a specified definition.
+   *
+   * @param name
+   *   the name of the index to be created
+   * @param definition
+   *   the settings for the index
+   * @return
+   *   an instance of [[CreateIndexRequest]] that represents the create index operation to be performed.
+   */
   final def createIndex(name: IndexName, definition: String): CreateIndexRequest =
     CreateIndex(name = name, definition = Some(definition))
 
+  /**
+   * Constructs an instance of [[DeleteByIdRequest]] used for deleting a document from the specified index by specified
+   * ID.
+   *
+   * @param index
+   *   the name of the index to delete the document from
+   * @param id
+   *   the ID of the document to be deleted
+   * @return
+   *   an instance of [[DeleteByIdRequest]] that represents delete by id operation to be performed.
+   */
   final def deleteById(index: IndexName, id: DocumentId): DeleteByIdRequest =
     DeleteById(index = index, id = id, refresh = None, routing = None)
 
+  /**
+   * Constructs an instance of [[DeleteByQueryRequest]] used for deleting documents from the specified index by
```

Comment:
Maybe - deleting documents from the specified index that satisfy specified query